vignettes/use-application-audit.Rmd

```diff
@@ -0,0 +1,492 @@
+
+  # Treat data and generate report
+  df <- shinyloadtest::load_runs("run1")
+  shinyloadtest::shinyloadtest_report(
+    df,
+    "public/index.html",
+    self_contained = TRUE,
+    open_browser = FALSE
+  )
+}
+```
+
+Below is the necessary GitHub Actions yaml file. Overall, this will:
+
+- Run each time code is pushed to GitHub.
+- Run in parallel on 3 different Ubuntu flavors.
+- Install R, some system dependencies and R packages. Install Chrome and 
+shinycannon tools for the headless loadtest.
+- Deploy the load test report to GitHub Pages.
+
+```yaml
+on:
+  push:
+    branches: [main, master]
+  pull_request:
+    branches: [main, master]
+
+name: shiny-loadtest-ci
+
+jobs:
+  shiny-loadtest-ci:
+    runs-on: ${{ matrix.config.os }}
+    permissions:
+      contents: write
+
+    name: ${{ matrix.config.os }} (${{ matrix.config.r }})
+
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {os: ubuntu-latest,   r: 'devel', http-user-agent: 'release'}
+          - {os: ubuntu-latest,   r: 'release'}
+          - {os: ubuntu-latest,   r: 'oldrel-1'}
+
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      R_KEEP_PKG_SOURCE: yes
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: r-lib/actions/setup-r@v2
+      - uses: r-lib/actions/setup-r-dependencies@v2
+        with:
+          cache-version: 2
+          extra-packages: |
+            any::shinyloadtest
+            any::lubridate
+            any::DT
+            any::callr
+            any::shinytest2
+            any::deSolve
+            any::attempt
+
+      - name: Install shinycannon 💥 and Chrome 🌐
+        run: |
+          sudo bash -c 'apt-get update; apt-get install -y default-jre-headless; apt-get clean; rm -rf /var/lib/apt/lists/*'
+          wget https://github.com/rstudio/shinycannon/releases/download/v1.1.3/shinycannon_1.1.3-dd43f6b_amd64.deb https://dl.google.com/linux/direct/google-chrome-stable_current_amd64.deb
+          sudo dpkg -i ./*.deb
+
+      - name: Run load test 🏥
+        shell: Rscript {0}
+        run: source("R/audit-app.R"); record_loadtest(path = "app.R");
+
+      - name: Deploy to GitHub pages 🚀
+        if: github.event_name != 'pull_request'
+        uses: JamesIves/github-pages-deploy-action@4.1.4
+        with:
+          clean: false
+          branch: gh-pages
+          folder: public
+```
+
+If you want to test it on your end, below are lines of code to setup a 
+RStudio project and link it to github:
+
+``` r
+path <- tempdir()
+usethis::create_project(path, open = FALSE)
+# Inside the project
+workflows_path <- file.path(path, ".github/workflows")
+r_path <- file.path(path, "R")
+dir.create(workflows_path, recursive = TRUE)
+dir.create(r_path)
+# Copy all R helpers
+file.create(file.path(r_path, "audit-app.R"))
+# Copy the previous yaml template inside this file
+file.create(file.path(workflows_path, "shiny-loadtest-ci.yaml"))
+# copy your app content
+file.create("app.R") 
```

Comment:
This would be great if it could be something that could _almost_ be copy/paste-able. Something like...

```r
# Inside the project
# Use audit GitHub Actions workflow
usethis::use_github_action(url = "https://raw.githubusercontent.com/rstudio/shinytest2/main/.github/workflows/app-audit.yaml")
# Copy audit script
dir.create(file.path(path, "R"))
file.copy(system.file("gha/audit-app.R", package = "shinytest2"), file.path(path, "R/audit-app.R"))
# Create your app.R file
file.create("app.R") 

## Need to copy in your app code to `app.R`!
```